index.md

```diff
@@ -1453,14 +1448,415 @@ def check(condition, message):
 
     Note that it is **strongly recommended** that you [**always use four spaces per indentation level**][PEP8: Indentation] throughout your code.
 
-2.  *IndentationError: unexpected indent*
+5.  *IndentationError: unexpected indent*
 
     In Python, the only time you would increase the indentation level of your code is to define a new code block after a [compound statement][Compound statements] such as `for`, `if`, `def`, or `class`.
 
     ~~~~ {include="unexpected_indent"}
     ~~~~
 
 
+## New errors {#new}
+
+### Consider iterating dictionary (C0201) {#C0201}
+
+It is more *pythonic* to iterate through a dictionary directly, without calling the `.keys` method.
+
+~~~~ {include="C0201_consider_iterating_dictionary"}
+~~~~
+
+Corrected version:
+
+```python
+for item in menu:
+    print("My store sells {}.".format(item))
+```
+
+### Superfluous parens (C0325) {#C0325}
+
+This error occurs when a keyword, such as `if` or `for`, is followed by a single item enclosed in parentheses. In such a case, parentheses are not necessary.
+
+~~~~ {include="C0325_superfluous_parens"}
+~~~~
+
+Corrected version:
+
+```python
+if 'anchovies' in pizza_toppings:  # Error on this line
+    print("Awesome!")
+```
+
+### Literal comparison (R0123) {#R0123}
+
+This error occurs when we use the identity operator `is` to compare Python literals. Whether or not two literals representing the same value (e.g. two identical strings) have the same identity can vary, depending on the way the code is being executed, the code that has ran previously, and the version and implementation of the Python interpreter. For example, each of the following assertions pass if the lines are evaluated together from a Python file, but `assert num is 257` and `assert chars is 'this string fails'` fail if the lines are entered into a Python interpreter one-by-one.
+
+~~~~ {include="R0123_literal_comparison"}
+~~~~
+
+To prevent the confusion, it is advisable to use the equality operator `==` when comparing objects with Python literals.
+
+```python
+num = 256
+assert num == 256
+
+num = 257
+assert num == 257  # Assertion fails if typed into a Python interpreter
+
+chars = 'this_string_passes'
+assert chars == 'this_string_passes'
+
+chars = 'this string fails'
+assert chars == 'this string fails'  # Assertion fails if typed into a Python interpreter
+```
+
+**See also**:
+
+- [Literally Literals and Other Number Oddities In Python]
+- [StackOverflow: About the changing id of an immutable string]
+- [StackOverflow: When does Python allocate new memory for identical strings?]
+
+### Unsupported assignment operation (E1137) {#E1137}
+
+This error occurs when we assign something to an object which does not support assignment (i.e. an object which does not define the `__setitem__` method).
```

Comment:
I realized I don't like the code example or description here. I want you to be explicit about *attribute* assignment as the issue -- slicing introduces an extra level of complexity that's unnecessary. I imagine the following code example could work better:

```python
x = 3
x.new_attr = 'hello'  # Error on this line
```

You don't need to include a description of how to "fix" this problem.

Reply:
As far as I understand, this error only works for item assignment? I looked at the [examples pylint uses to test their code](https://github.com/PyCQA/pylint/blob/master/pylint/test/functional/unsupported_assignment_operation.py), and all of them are item assignment. I don't think the example you give throws this error. I did remove the code for how to "fix" the problem.

Reply:
No you're right, my bad. I was confusing `__setitem__` with `__setattr__`.